vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -4242,12 +4258,33 @@ private OverlayInfo createOverlayInfo(SheetOverlayWrapper overlayWrapper) {
 
         Sheet sheet = getActiveSheet();
 
-        int col = overlayWrapper.getAnchor().getCol1();
+        var anchor = overlayWrapper.getAnchor();
+
+        // Need special handling for XSSFClientAnchor anchors of type
+        // DONT_MOVE_AND_RESIZE.
+        // See https://github.com/vaadin/flow-components/issues/3261
+        if (AnchorType.DONT_MOVE_AND_RESIZE.equals(anchor.getAnchorType())
+                && anchor instanceof XSSFClientAnchor) {
+            info.col = 1;
+            info.row = 1;
+
+            var xssfAnchor = (XSSFClientAnchor) anchor;
+            info.dx = (Long) xssfAnchor.getPosition().getX()
+                    / Units.EMU_PER_PIXEL;
+            info.dy = (Long) xssfAnchor.getPosition().getY()
+                    / Units.EMU_PER_POINT;
+            info.width = xssfAnchor.getSize().getCx() / Units.EMU_PER_PIXEL;
+            info.height = xssfAnchor.getSize().getCy() / Units.EMU_PER_POINT;
```

Comment:
`EMU_PER_PIXEL` used for x offset and width, same as [here](https://github.com/vaadin/flow-components/blob/5876423ab1e14d4888ea5f8d800c6bfca5e0d6f1/vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/SheetOverlayWrapper.java#L96)

`EMU_PER_POINT` used for y offset and height, same as [here](https://github.com/vaadin/flow-components/blob/5876423ab1e14d4888ea5f8d800c6bfca5e0d6f1/vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/SheetOverlayWrapper.java#L130)